cloud_grafana_api_key.go

```diff
@@ -0,0 +1,61 @@
+package gapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// This function creates a API key inside the Grafana instance running in stack `stack`. It's used in order
+// to provision API keys inside Grafana while just having access to a Grafana Cloud API key.
+//
+// See https://grafana.com/docs/grafana-cloud/api/#create-grafana-api-keys for more information.
+func (c *Client) CreateGrafanaAPIKeyFromCloud(stack string, input *CreateAPIKeyRequest) (*CreateAPIKeyResponse, error) {
+	resp := CreateAPIKeyResponse{}
+	data, err := json.Marshal(input)
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.request("POST", fmt.Sprintf("/api/instances/%s/api/auth/keys", stack), nil, bytes.NewBuffer(data), &resp)
+	return &resp, err
```

Comment:
Nit: move the creating of `resp` closer to where it's used:
```suggestion
	data, err := json.Marshal(input)
	if err != nil {
		return nil, err
	}

	resp := &CreateAPIKeyResponse{}
	err = c.request("POST", fmt.Sprintf("/api/instances/%s/api/auth/keys", stack), nil, bytes.NewBuffer(data), resp)
	return resp, err
```

(Note that it also creates it as a pointer variable)